azurelinuxagent/agent.py

```diff
@@ -183,11 +200,14 @@ def main(args=[]): # pylint: disable=R0912,W0102
                 agent.run_exthandlers(debug)
             elif command == "show-configuration":
                 agent.show_configuration()
+            elif command == "collect-logs":
```

Comment:
Maybe a bit tangential to this PR, but could we also make an Enum of all these commands just to make the code a bit cleaner. Thanks!

Reply:
I agree, the entire `agent.py` module seems to be a bit wacky. I will have to defer this change for later, though :)